src/azure-cli/azure/cli/command_modules/role/custom.py

```diff
@@ -588,7 +588,7 @@ def list_apps(cmd, app_id=None, display_name=None, identifier_uri=None, query_fi
 
     result = client.applications.list(filter=(' and '.join(sub_filters)))
     if sub_filters or include_all:
-        return result
+        return list(result)
```

Comment:
`result` returned by `client.applications.list` is an `Iterator`. It is later used by https://github.com/Azure/azure-cli/blob/50ccb24528eb36529d1432808f3ff385cc4c04aa/src/azure-cli/azure/cli/command_modules/role/custom.py#L782

which will always evaluate to `True`.

`result` should be converted to a `list` before being evaluated by `if`.

Example:

```py
it = iter([])
print(bool(it))
# True
```